.github/actions/test-setup/action.yaml

```diff
@@ -0,0 +1,44 @@
+name: "Test setup"
+inputs:
+  node-version:
+    description: "Node version"
+    default: 18.14.1
+  pnpm-version:
+    description: "PNPM version"
+    default: 8.9.0
+# TODO: Implement when #1310 has been merged
+# inputs:
+#   checkout-fetch-depth:
+#     description: "Number of commits to fetch. 0 indicates all history for all branches and tags."
+#     default: 1
+runs:
+  using: "composite"
+  steps:
+    # TODO: Implement when #1310 has been merged
+    # - name: Checkout
+    #   uses: actions/checkout@v3
+    #   with:
+    #     fetch-depth: ${{ inputs.checkout-fetch-depth }}
```

Comment:
@camsjams had requested that we moved the test setup jobs to a composite action. One stage of that action would be to call the checkout job. We can't use a composite action that is not on `master` to then checkout to the branch in PR, until the composite action itself is on `master`. This PR will introduce the action, so that checkout can then be called from it in a further iteration. Otherwise we need to keep checkouts in on the top level workflow, i.e. `test.yaml`.

Reply:
Extracted in #1389